streamz/tests/test_core.py

```diff
@@ -699,6 +699,20 @@ def test_unique_key():
     assert L == [1, 2, 3]
 
 
+def test_unique_metadata():
+    @Stream.register_api()
+    class get_metadata(Stream):
+        def update(self, x, metadata=None, who=None):
+            self._emit(metadata)
+
+    source = Stream()
+    L = source.unique().get_metadata().flatten().sink_to_list()
```

Comment:
Is the metadata not already flat? All functions should be forwarding a flat list.

Reply:
```
>>> from streamz import Stream
>>> from streamz.utils_test import metadata
>>> src = Stream()
>>> L = metadata(src).sink_to_list()
>>> src.emit(0, metadata=[{'a': 1}])
>>> src.emit(1, metadata=[{'b': 2}])
>>> L
[[{'a': 1}], [{'b': 2}]]
```
As far as I understood [the docs](https://github.com/python-streamz/streamz/blob/master/docs/source/core.rst#metadata), metadata should always be a list. In this situation, the result of `metadata(...)` is a list of lists.

Reply:
Oh yes, I see. There is one element on the output for each emit.